Can not assign bidirectional to forward any_view

Consider the following:

```
int main()
{
    map<string, string> mm{ make_pair("hi", "there"), make_pair("good", "bye") };
    auto just_keys = mm | view::keys;
    v3::any_view<string, v3::category::forward | v3::category::sized> as_any = just_keys;
    
    cout << "The keys: " << as_any;
}
```

It fails with:

```
/opt/wandbox/range-v3/include/range/v3/view/any_view.hpp:325:56: error: non-virtual member function marked 'override' hides virtual member function
                bool equal(Forward const &that_) const override
```

But works fine if my `any_view` is declared with concept bidirectional. I know that the view returned by `keys` is bidirectional but I think I should be allowed to assign it a any_view with strictly fewer requirements.

Live demo: https://wandbox.org/permlink/I3Cflj0MCkOi5lOS

I'm fairly new to ranges so perhaps I'm just thinking about this wrong. My use case is that I have an interface that returns an `any_view` that's `sized` and `forward`. I have a subclass that overrides that method and it seems to me that there should be some way to convert my bidirectional range into a forward one so I can correctly override the base class method.

--- a/include/range/v3/view/any_view.hpp
+++ b/include/range/v3/view/any_view.hpp
@@ -50,44 +50,44 @@ namespace ranges
          *  \relates ranges::v3::category
          *  \{
          */
-        constexpr category operator& (category lhs, category rhs) noexcept
+        constexpr category operator&(category lhs, category rhs) noexcept
         {
             return static_cast<category>(
                 static_cast<meta::_t<std::underlying_type<category>>>(lhs) &
                 static_cast<meta::_t<std::underlying_type<category>>>(rhs));
         }
 
-        constexpr category operator| (category lhs, category rhs) noexcept
+        constexpr category operator|(category lhs, category rhs) noexcept
         {
             return static_cast<category>(
                 static_cast<meta::_t<std::underlying_type<category>>>(lhs) |
                 static_cast<meta::_t<std::underlying_type<category>>>(rhs));
         }
 
-        constexpr category operator^ (category lhs, category rhs) noexcept
+        constexpr category operator^(category lhs, category rhs) noexcept
         {
             return static_cast<category>(
                 static_cast<meta::_t<std::underlying_type<category>>>(lhs) ^
                 static_cast<meta::_t<std::underlying_type<category>>>(rhs));
         }
 
-        constexpr category operator~ (category lhs) noexcept
+        constexpr category operator~(category lhs) noexcept
         {
             return static_cast<category>(
                 ~static_cast<meta::_t<std::underlying_type<category>>>(lhs));
         }
 
-        RANGES_CXX14_CONSTEXPR category & operator&= (category & lhs, category rhs) noexcept
+        RANGES_CXX14_CONSTEXPR category &operator&=(category &lhs, category rhs) noexcept
         {
             return (lhs = lhs & rhs);
         }
 
-        RANGES_CXX14_CONSTEXPR category & operator|= (category & lhs, category rhs) noexcept
+        RANGES_CXX14_CONSTEXPR category &operator|=(category &lhs, category rhs) noexcept
         {
             return (lhs = lhs | rhs);
         }
 
-        RANGES_CXX14_CONSTEXPR category & operator^= (category & lhs, category rhs) noexcept
+        RANGES_CXX14_CONSTEXPR category &operator^=(category &lhs, category rhs) noexcept
         {
             return (lhs = lhs ^ rhs);
         }
@@ -281,14 +281,14 @@ namespace ranges
 
             template<typename Ref, category Cat>
             struct any_cursor_interface<Ref, Cat, meta::if_c<(Cat & category::mask) == category::bidirectional>>
-              : any_cursor_interface<Ref, category::forward>
+              : any_cursor_interface<Ref, (Cat & ~category::mask) | category::forward>
             {
                 virtual void prev() = 0;
             };
 
             template<typename Ref, category Cat>
             struct any_cursor_interface<Ref, Cat, meta::if_c<(Cat & category::mask) == category::random_access>>
-              : any_cursor_interface<Ref, category::bidirectional>
+              : any_cursor_interface<Ref, (Cat & ~category::mask) | category::bidirectional>
             {
                 virtual void advance(std::ptrdiff_t) = 0;
                 virtual std::ptrdiff_t distance_to(any_cursor_interface const &) const = 0;
@@ -310,7 +310,7 @@ namespace ranges
                   : it_{std::move(it)}
                 {}
             private:
-                using Forward = any_cursor_interface<Ref, category::forward>;
+                using Forward = any_cursor_interface<Ref, (Cat & ~category::mask) | category::forward>;
 
                 I it_;
 

--- a/test/view/any_view.cpp
+++ b/test/view/any_view.cpp
@@ -9,9 +9,11 @@
 //
 // Project home: https://github.com/ericniebler/range-v3
 
+#include <map>
 #include <vector>
 #include <range/v3/core.hpp>
 #include <range/v3/view/iota.hpp>
+#include <range/v3/view/map.hpp>
 #include <range/v3/view/take.hpp>
 #include <range/v3/view/reverse.hpp>
 #include <range/v3/view/any_view.hpp>
@@ -170,6 +172,14 @@ RANGES_DIAGNOSTIC_POP
             ten_ints.begin(), std::ptrdiff_t(ten_ints.size())
         }};
         ::check_equal(v, ten_ints);
+    }
+
+    // Regression test for #880
+    {
+        std::map<int, int> mm{ {0, 1}, {2, 3} };
+        ranges::any_view<int, ranges::category::forward | ranges::category::sized> as_any =
+            mm | ranges::view::keys;
+        (void)as_any;
     }
 
     test_polymorphic_downcast();

